### PR TITLE
NucliaDB tests unification: writer

### PR DESCRIPTION
**Pull request by jotare**
### Description
- Port writer tests to ndbfixtures
- Add a new magic fixture for component `nucliadb_writer`
- Add more typing to the tests

### How was this PR tested?
Tunned tests

### CODE REVIEW

File: nucliadb/tests/conftest.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 #
 pytest_plugins = [
+    "pytest_mock",
```
Comment (javitonino) [inline nucliadb/tests/conftest.py:21]: Can we remove the `conftest.py` in the `writer` directory now?

File: nucliadb/tests/ndbfixtures/__init__.py [hunk 2]
```diff
@@ -18,19 +18,28 @@
...
 
 # components and deployment modes
 
 from .reader import *  # noqa
+from .writer import *  # noqa
 
 # subcomponents
 
-from .common import *  # noqa
+from .common import *  # type: ignore # noqa
```
Comment (javitonino) [inline nucliadb/tests/ndbfixtures/__init__.py:39]: Why? :thinking: I guess some things are getting overridden? In that case, could we not just stop doing `import *` of all  the things?

File: nucliadb/tests/ndbfixtures/writer.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    async with client_factory(roles=[NucliaDBRoles.WRITER]) as client:
+        yield client
+
+
+@pytest.fixture(scope="function")
+async def nucliadb_writer_manager(
+    nucliadb_writer: AsyncClient,
+) -> AsyncIterator[AsyncClient]:
+    roles = [NucliaDBRoles.MANAGER, NucliaDBRoles.WRITER]
+    nucliadb_writer.headers["X-NUCLIADB-ROLES"] = ";".join([role.value for role in roles])
```
Comment (javitonino) [inline nucliadb/tests/ndbfixtures/writer.py:64]: Could we use the `client_factory` for this fixture as well? It feels a bit cleaner than manipulating headers directly.

File: nucliadb/tests/ndbfixtures/writer.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    storage_driver = request.param
+    set_utility(Utility.STORAGE, storage_driver)
+
+    yield storage_driver
+
+    clean_utility(Utility.STORAGE)
+
+
+# FIXME: this is a weird situation, we can use a hosted-like nucliadb while this
+# creates a KB as it was onprem. The end result should change much but still, is
```
Comment (javitonino) [inline nucliadb/tests/ndbfixtures/writer.py:145]:
```suggestion
# creates a KB as it was onprem. The end result should not change much but still, is
```

**Review by javitonino — APPROVED** [review]
Some questions/comments, but looks good!

File: nucliadb/tests/ndbfixtures/writer.py (hunk 3 above)
Comment (jotare) [reply nucliadb/tests/ndbfixtures/writer.py:64]: I'd prefer not to. Maybe it's not explicit enough but this is using a magic fixture `nucliadb_writer`, that is an AsyncClient to the component, standalone or whatever. It doesn't need to know

File: nucliadb/tests/conftest.py (hunk 1 above)
Comment (jotare) [reply nucliadb/tests/conftest.py:21]: Yeah, I'll change to a more explicit approach in conftest and avoid this hack

File: nucliadb/tests/ndbfixtures/__init__.py (hunk 2 above)
Comment (jotare) [reply nucliadb/tests/ndbfixtures/__init__.py:39]: Some hack I did in the past. I'll change to a more explicit approach in conftest and avoid this hack

File: nucliadb/tests/conftest.py (hunk 1 above)
Comment (jotare) [reply nucliadb/tests/conftest.py:21]: It is deleted. This is a general conftest for all migrated packages (currently reader and writer)

Comment (javitonino) [reply nucliadb/tests/conftest.py]: Uhm, I don't see it deleted in this PR. :confused:

**Pull request merged**